### PR TITLE
Object stripping causes issues 

### LINKED ISSUE #40: Issue with empty sub-objects
Deserialzing 

```json
{
                "resourceType": "ClinicalImpression",
                "status": "completed",
                "subject": {"reference": ""},
}
```

Leads to a internal representation of:
```
resource_type='ClinicalImpression' id=None meta=None implicit_rules=None language=None text=None contained=None extension=None modifier_extension=None identifier=None status=<EventStatus.completed: 'completed'> status_reason=None code=None description=None subject=Reference(id=None, extension=None, reference=None, type=None, identifier=None, display=None) encounter=None effective_date_time=None effective_period=None date=None assessor=None previous=None problem=None investigation=None protocol=None summary=None finding=None prognosis_codeable_concept=None prognosis_reference=None supporting_info=None note=None
```

Which then leads to a serialization of, which is an invalid `ClinicalImpression` since `subject` has a carnality of `1...1`:

```json
{
                "resourceType": "ClinicalImpression",
                "status": "completed"
}
```

This behavior is half correct. The subject should actually be stripped before de-serialization, which then would lead to a `ValidationError`, because `subject` is a required field.

### PR DESCRIPTION
**Pull request by julian-r**

### CODE REVIEW

**Review by auerj — APPROVED** [review]

File: tests/pydantic/test_nullability.py [hunk 1]
```diff
@@ -68,26 +68,22 @@ def test_empty_lists_are_ignored():
     assert RootModel(field_a="a", list_items=[]).dict() == {
         "field_a": "a",
     }
-    assert RootModel(
-        field_a="a", list_items=[ListItem(), ListItem(), ListItem()]
-    ).dict() == {"field_a": "a"}
+    with pytest.raises(pydantic.ValidationError):
```
Comment (trojanowski) [inline tests/pydantic/test_nullability.py:71]: I'd also leave a variant with `ListItem()` to ensure we have a validation error there. We should change the name of this test.

File: tests/pydantic/test_nullability.py [hunk 2]
```diff
@@ -68,26 +68,22 @@ def test_empty_lists_are_ignored():
...
-        ],
-    ).dict() == {
-        "field_a": "a",
-        "list_items": [
-            {"list_field_a": "list item 1 a"},
-            {"list_field_a": "list item 3 a"},
-        ],
-    }
+    with pytest.raises(pydantic.ValidationError):
+        RootModel(
```
Comment (trojanowski) [inline tests/pydantic/test_nullability.py:77]: The same as above - I'd add a check for usage with class instances and rename the test.

File: tests/pydantic/test_nullability.py (hunk 1 above)
Comment (julian-r) [reply tests/pydantic/test_nullability.py:71]: Agree. I created an extra issue(#42) to address this.

File: tests/pydantic/test_nullability.py (hunk 2 above)
Comment (julian-r) [reply tests/pydantic/test_nullability.py:77]: Agree. I created an extra issue(#42) to address this.

**Pull request merged**